r2dbc-spec/src/main/asciidoc/data-types.adoc

```diff
@@ -104,11 +109,15 @@ Driver implementations should implement the following type mapping and can suppo
 | `java.math.BigDecimal`
 
 | `FLOAT(p)`
-| Represents an approximate numerical with mantissa precision `p`.
-| `java.lang.Double`
+| Represents an approximate numerical with mantissa precision `p`. Databases using IEEE representation can map values to either 32-bit or 64-bit floating point types depending on precision parameter `p`.
+| `java.lang.Double` or `java.lang.Float`
 
 | `REAL`
 | Same as `FLOAT` type except that the DBMS defines the precision.
+| `java.lang.Float`
+
+| `DOUBLE PRECISION`
+| Represents an approximate numerical.
```

Comment:
`REAL` and `DOUBLE PRECISION` need to have similar descriptions. `REAL` is an approximate numeric data type with some database-specific precision. `DOUBLE PRECISION` is an approximate numeric data type with some database-specific precision that is greater than precision of `REAL` data type.

Many databases handle them as 32-bit and 64-bit IEEE binary floating point numbers.